device_backends/include/AsyncDomainImpl.h

```diff
@@ -47,7 +47,8 @@ namespace ChimeraTK {
      *  @ return The version number that has been used for distribution.
      */
     VersionNumber activate(BackendDataType data, VersionNumber version = VersionNumber{nullptr});
-    void deactivate();
+
+    void deactivate() override;
     void sendException(const std::exception_ptr& e) noexcept override;
 
     template<typename UserDataType>
```

Comment:
`subscribe()` could become a virtual function as well (with the ususal trick for UserType-templated virtual functions), which might be helpful for backends that have varying `BackendDataType`. For those backends (maybe even generally for all backends) I would implement some kind of "createOrGetAsyncDomain" function, in which case the calling code (e.g. `getRegisterAccessor()`) does not need to know the backend type.

Reply:
I agree that ```subscribe()``` could become a UserType-templated virtual functions. However, debugging those is a pain in the a.... The compiler error messages turn in circles and rarely point to the line that has the actual error, and stepping through with the debugger is very tedious. I would only use them where it is absolutely necessary.

I introduced ```getOrCreateAsyncDomain()``` in the ```AsyncDomainsContainer```, which is returning a shared pointer to the implementation, not to the base class. The function needs to be templated to the backend because it internally has to get the initial value from the backend, which is of the backend specific user type and might vary from domain to domain. The calling code has to provide this information to ``getOrCreateAsyncDomain()``` as template parameter, which is always called before the ```subscribe()``` call, so there is no benefit from adding  UserType-templated virtual functions for ```subscribe()```.